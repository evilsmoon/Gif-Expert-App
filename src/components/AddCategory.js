import React, { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {


    const [inputValue, setinputValue] = useState('');
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
        // console.log('handleInputChange llamado');
    }
    const handelSubmit = (e) => {
        e.preventDefault();
        console.log('handelSubmit llamado ', inputValue);
        
        if(inputValue.trim().length >2){
            setCategories(c => [inputValue,...c]);
            setinputValue('');
        }
    }
    return (

        <form onSubmit={ handelSubmit }>  
            <p>{ inputValue }</p>
            <input
                type='text'
                value={ inputValue }
                onChange={ handleInputChange }
            />

        </form>
    )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
