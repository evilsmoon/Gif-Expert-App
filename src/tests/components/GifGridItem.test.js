import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

// import  from '../../components/GifGridItem'
describe('Test GifGridItem.js ', () => {
    // let wrapper = shallow(<GifGridItem/>);
    // beforeEach(()=>{
        // wrapper = shallow(<GifGridItem/>)
    // })
    const title = 'titulo';   
    const url = 'https://localhost/image.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('mostrar componente <GifGridItem/>  ', () => {


        expect(wrapper).toMatchSnapshot();

        
    })
    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)    
    })
    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props('src'))

        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    })
    
    test('debe de tener animate__fadeIn ', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes('animate__fadeIn')).toBe( true)
    })
    
            
    
})
