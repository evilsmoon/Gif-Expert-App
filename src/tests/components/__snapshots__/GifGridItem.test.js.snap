// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem.js  mostrar componente <GifGridItem/>   1`] = `
<div
  className="card animate__animated animate__fadeIn  "
>
  <img
    alt="titulo"
    src="https://localhost/image.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
